File: Simple Factory/Logger/Python/LoggerFactory.py
from DebugLogger import DebugLogger
from ErrorLogger import ErrorLogger
from InfoLogger import InfoLogger
from LogLevel import LogLevel
from ILogger import ILogger
class LoggerFactory:
    @staticmethod
    def createLogger(logLevel: LogLevel) -> ILogger | None:
        match logLevel:
            case "DEBUG":
                return DebugLogger()
            case "ERROR":
                return ErrorLogger()
            case "INFO":
                return InfoLogger()
            case _:
                return None
            

File: Simple Factory/Logger/JAVA/LoggerFactory.java
public class LoggerFactory {
    public static ILogger createLogger(LogLevel logLevel) {
        switch(logLevel) {
            case DEBUG:
                return new DebugLogger();
            case ERROR:
                return new ErrorLogger();
            case INFO:
                return new InfoLogger();
            default:
                return null;

        }
    }
}